### PR TITLE
[INITIAL] Adding Syslog export plugin

### PR DESCRIPTION
**Pull request by Karthick-Somasundaresan**
This plugin will send the syslog messages from
WPEFramework over websocket.

### CODE REVIEW

**Comment by mhughesacn** [comment]
Hi Karthick - Please use a Metrological copyright in your headers, not RDK, to fix the blackduck check.
Martin Hughes, RDK CMF Compliance Team

**Comment by Karthick-Somasundaresan** [comment]
> Hi Karthick - Please use a Metrological copyright in your headers, not RDK, to fix the blackduck check.
> Martin Hughes, RDK CMF Compliance Team

Thank you. Will correct it.

File: SyslogExport/SyslogExport.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        return;
+    }
+
+    bool SyslogExport::WebsocketSink::AddChannel(WPEFramework::PluginHost::Channel& channel)
+    {
+        bool accepted = false;
+        _lock.Lock();
+
+        if ((_maxConnections !=0) && (_maxConnections - _exportChannelMap.size() > 0)) {
+            ASSERT(0 == _exportChannelMap.count(channel.Id()));
```
Comment (sramani-metro) [inline SyslogExport/SyslogExport.cpp:161]: Does this have to be an assert? You could re-add?

Comment (Karthick-Somasundaresan) [reply SyslogExport/SyslogExport.cpp:161]: Channel ids are unique. Ideally, the same channel number should not come twice. If we are already having a channel id in the map then there is some problem. Hence we are doing an Assert.

File: SyslogExport/SyslogExport.h [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+#pragma once
+
+#include "Module.h"
+#include <map>
+#include <tuple>
```
[the remark this answers is not in the document]
Comment (Karthick-Somasundaresan) [reply SyslogExport/SyslogExport.h:24]: No, I shall remove it.

File: SyslogExport/SyslogExport.h [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            std::queue<std::string> _logList;
+            std::mutex _mutex;
+            std::condition_variable _condVar;
+            std::unique_ptr<SyslogExport::WebsocketSink> _websocketSink;
+
+        };
+
+
+    private:
+        uint32_t _connectionId;
```
[the remark this answers is not in the document]
Comment (Karthick-Somasundaresan) [reply SyslogExport/SyslogExport.h:188]: No, I shall remove it.

File: SyslogExport/SyslogExport.h [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        SinkManager _sinkManager;
+        Notification _notification;
+        Config _config;
```
[the remark this answers is not in the document]
Comment (Karthick-Somasundaresan) [reply SyslogExport/SyslogExport.h:191]: Agreed. I shall make the change accordingly.

**Comment by MFransen69** [comment]
Discussed this with Karthick. He will create an even improved version of this. Until then I will close the PR

**Pull request closed without merging**